src/Range.tsx

```diff
@@ -327,6 +327,20 @@ class Range extends React.Component<IProps> {
       disabled
     } = this.props;
     const { draggedThumbIndex } = this.state;
+
+    const zIndexes = new Array(values.length).fill(0);
+    const lenValues = values.length;
+
+    // Only need special handling of zIndexes when there are multiple values
+    // and at least one value (the last one) is at the max value
+    if (lenValues > 1 && values[lenValues - 1] === max && values[lenValues - 2] === max) {
```

Comment:
It should also check for

```js
 !this.props.allowOverlap
```

if overlap is allowed then there is no need for this functionality